src/lib/incapsula.ts

```diff
@@ -0,0 +1,45 @@
+import * as bluebird from 'bluebird';
+import * as http from 'superagent';
+import { buildServiceResponse, ServiceResponse } from '../lib/service';
+
+const BASE_URL = 'https://my.incapsula.com';
+const BASE_PATH = '/api/prov/v1';
+
+export function flushSite(siteID: string, incapsulaApiID: string, incapsulaApiKey: string): bluebird<ServiceResponse> {
+  return buildServiceResponse(
+    http
+      .post(
+        `${BASE_URL}${BASE_PATH}/sites/cache/purge?api_id=${incapsulaApiID}&api_key=${incapsulaApiKey}&site_id=${siteID}`
```

Comment:
I think this is safer to put parameters into the body (ie. they will not be logged anywhere) of the POST request (this is what the Incapsula documentation recommend).

Reply:
I think you refer to the v2 version of the API. I have followed the documentation of the v1 that uses query param.
https://docs.imperva.com/bundle/cloud-application-security/page/api/sites-api.htm#Purge

Reply:
Why don't we use the v2 of the API?

Reply:
Because that looked WIP to me. We discussed that with @abarre

Reply:
But we can migrate to v2 if it's safer

Reply:
In the API v1 documentation, parameters are passed into the POST request body:
https://docs.imperva.com/bundle/cloud-application-security/page/api/api.htm#Overview

Reply:
right, I guess I get confused by the exemple given on this page:
https://docs.imperva.com/bundle/cloud-application-security/page/api/sites-api.htm#Purge
```
https://docs.imperva.com/bundle/cloud-application-security/page/api/sites-api.htm#Purge 
ie: curl -X POST "https://my.imperva.com/api/prov/v1/sites/incapRules/list?api_key=48d69342-eaec-44cf-8a5c-56c4ff1cd5e8&api_id=12593&site_id=12345"

```

Reply:
but again, if it works and we think it's better to post param in the body, I can make those changes accordingly

Reply:
I think it is preferable to "hide" parameters into the POST body

Reply:
There is a misunderstanding here. 
There is no incapsula API to flush assets by hostname.

The only thing mentioning hostname is this part in the doc related to internal DNS cache :

> Purge hostname from cache
> Use this operation to purge the hostname from the cache.

> This API is for customers who use the same CNAME provided by Imperva for multiple hostnames and would like to change the CNAME for a particular hostname. Purging the hostname is required for the CNAME change to take effect.

> /api/prov/v1/sites/hostname/purge

Reply:
So, I suggest to simply remove this route.